near-plugins/tests/contracts/pausable/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+            counter: 0,
+            __acl: Default::default(),
+        };
+
+        // Make the contract itself super admin. This allows us to grant any role in the
+        // constructor.
+        near_sdk::require!(
+            contract.acl_init_super_admin(env::predecessor_account_id()),
+            "Failed to initialize super admin",
+        );
+
+        // Grant `Role::PauseManager` to the provided account.
+        let result = contract.acl_grant_role(Role::PauseManager.into(), pause_manager);
+        near_sdk::require!(Some(true) == result, "Failed to grant role");
+
+        contract
+    }
+
+    pub fn get_counter(&self) -> u64 {
+        self.counter
+    }
+
+    /// Function can be paused using feature name "increase_1" or "ALL" like:
+    /// `contract.pa_pause_feature("increase_1")` or `contract.pa_pause_feature("ALL")`
+    ///
+    /// If the function is paused, all calls to it will fail. Even calls initiated by accounts which
+    /// are access control super admin or role grantee.
+    #[pause]
+    pub fn increase_1(&mut self) {
+        self.counter += 1;
+    }
+
+    /// Similar to `#[pause]` but use an explicit name for the feature. In this case the feature to
+    /// be paused is named "Increase by two". Note that trying to pause it using "increase_2" will
+    /// not have any effect.
+    ///
+    /// This can be used to pause a subset of the methods at once without requiring to use "ALL".
+    #[pause(name = "Increase by two")]
+    pub fn increase_2(&mut self) {
+        self.counter += 2;
+    }
+
+    /// Similar to `#[pause]` but roles passed as argument may still successfully call this method.
+    #[pause(except(roles(Role::Unrestricted4Increaser, Role::Unrestricted4Modifier)))]
+    pub fn increase_4(&mut self) {
+        self.counter += 4;
+    }
+
+    /// This method can only be called when "increase_1" is paused. Use this macro to create escape
+    /// hatches when some features are paused. Note that if "ALL" is specified the "increase_1" is
+    /// considered to be paused.
+    #[if_paused(name = "increase_1")]
+    pub fn decrease_1(&mut self) {
+        self.counter -= 1;
+    }
+
+    /// Similar to `#[if_paused]` but roles passed as argument may successfully call the method even
+    /// when the feature is _not_ paused.
+    #[if_paused(name = "increase_4", except(roles(Role::Unrestricted4Decreaser)))]
+    pub fn decrease_4(&mut self) {
+        self.counter -= 4;
+    }
+
+    /// Custom use of pause features. Only allow increasing the counter using `careful_increase` if
+    /// it is below 3.
+    pub fn careful_increase(&mut self) {
+        if self.counter >= 3 {
+            assert!(
+                !self.pa_is_paused("increase_big".to_string()),
+                "Method paused for large values of counter"
+            );
+        }
+
+        self.counter += 1;
+    }
+}
+
+/// Exposing internal methods to facilitate integration testing.
+/// TODO needed for this contract?
```

Comment:
Is this question answered now?

Reply:
It turned out that these functions are not needed for the `Pausable` test contract. They're removed in 5fe4f37.

Thanks for catching another open TODO!